extensions/amp-form/0.1/amp-form.js

```diff
@@ -488,14 +493,57 @@ export class AmpForm {
       dev().error(TAG, 'Form submit service failed: %s', e);
     }
 
+    // Get our special fields
     const varSubsFields = this.getVarSubsFields_();
-    if (this.xhrAction_) {
-      this.handleXhrSubmit_(varSubsFields, trust);
-    } else if (this.method_ == 'POST') {
-      this.handleNonXhrPost_();
-    } else if (this.method_ == 'GET') {
-      this.handleNonXhrGet_(varSubsFields);
+    const asyncInputs = toArray(
+        this.form_.querySelectorAll(
+            `.${escapeCssSelectorIdent(AsyncInputClasses['ASYNC_INPUT'])}`
+        )
+    );
+
+    // Do any assertions we may need to do
+    // For NonXhrGET
+    // took out all the code in handleNonXhrGet_
+    // because alot was assertions we want to fail on before getting var subs
+    if (!this.xhrAction_ && this.method_ == 'GET') {
```

Comment:
I don't understand why this change was made to `handleNonXhrGet`. Is it an optimization to avoid waiting for the the `Promise.all` call in case there are no `asyncInput`s? If that is the case, this `if` block is redundant because the `Promise.all` call should be a noop if it is passed an empty array.

Why do we want the assertions to fail before getting var subs?

Reply:
Yes, you mostly right 😄 

I moved this out in order to to be able to check all of the assertions for NonXHRGet. We want them to "fail" (bad choice of words on my part), so we can determine if we can submit synchronously vs async. 

So, while it is also an optimization, I mostly did this so that if we don't have any async inputs, then we know we can submit sync. Even though the Promise.all will noop, I am pretty sure that it will still be popped onto the task queue, thus making it async, and it won't synchronously run the `.then` block. Please correct me if I am wrong 😄

Reply:
Cool, yeah you're right about the task queue. If you keep this structure, does it still make sense for the non-XHR get to be in the block below?

https://github.com/ampproject/amphtml/blob/08720df6fabea59cc2a1f032b8fe46bb3374130d/extensions/amp-form/0.1/amp-form.js#L543-L545

Reply:
Rad!

And yes 😄 The blow above that calls `this.handleNonXhrGet_();` will handle the sync case. This will call `this.handleNonXhrGet_();` in the async case once we are done making all of our hidden inputs and things 😄